### PR TITLE
Stop using createImmutableMapReducer.

### PR DESCRIPTION
**Pull request by theengineear**
We decided that it was more confusing than helpful to include this
construct. It was originally included in Play where we had a highly-
nested structure. However, in core, we're going for a much flatter
structure where this doesn't really provide much else.

### CODE REVIEW

File: applications/jupyter-extension/nteract_on_jupyter/app/store.js [hunk 1]
```diff
@@ -49,6 +50,9 @@ const rootReducer = combineReducers({
   comms,
   config,
   modals,
+  communication: combineReducers({
+    kernelspecsByRef: communicationKernelspecsByRef
+  }),
```
Comment (theengineear) [inline applications/jupyter-extension/nteract_on_jupyter/app/store.js:55]: Eh, we're not using this yet, but might as well actually hook it up 😛

**Comment by theengineear** [comment]
@rgbkrk as we talked about :)

File: applications/jupyter-extension/nteract_on_jupyter/app/store.js (hunk 1 above)
Comment (rgbkrk) [reply applications/jupyter-extension/nteract_on_jupyter/app/store.js:55]: Works for me!

Comment (rgbkrk) [reply applications/jupyter-extension/nteract_on_jupyter/app/store.js:55]: It's fine iteration!

File: packages/core/src/reducers/kernelspecs.js [hunk 2]
```diff
@@ -4,59 +4,51 @@ import {
...
+        makeCommunicationKernelspecs({ loading: false, error: null })
+      );
     case actionTypes.FETCH_KERNELSPECS_FAILED:
-      return action.payload.error;
+      return state.set(
+        action.payload.kernelspecsRef,
+        makeCommunicationKernelspecs({
+          loading: false,
+          error: action.payload.error
+        })
```
Comment (rgbkrk) [inline packages/core/src/reducers/kernelspecs.js:32]: These are perfect and I can completely understand them in one quick read.

**Review by rgbkrk — APPROVED** [review]

File: packages/core/src/reducers/kernelspecs.js [hunk 3]
```diff
@@ -4,59 +4,51 @@ import {
...
+) => {
   switch (action.type) {
     case actionTypes.FETCH_KERNELSPECS_FULFILLED:
-      return Immutable.Map(action.payload.kernelspecs).map(
-        (kernelspec: KernelspecProps) => makeKernelspec(kernelspec)
+      return state.set(
+        action.payload.kernelspecsRef,
+        Immutable.Map(action.payload.kernelspecs).map(
+          (kernelspec: KernelspecProps) => makeKernelspec(kernelspec)
+        )
```
Comment (rgbkrk) [inline packages/core/src/reducers/kernelspecs.js:49]: 👌

Comment (rgbkrk) [reply packages/core/src/reducers/kernelspecs.js:49]: I'd probably leave comments in here that say

```
// This sets the kernelspecs as kernelspecsRef to contain
// a map of name -> KernelspecRecord
```

**Pull request merged**